rust/protocol/src/curve/curve25519.rs

```diff
@@ -36,6 +65,17 @@ impl KeyPair {
         }
     }
 
+    /// Do a [DH] key exchange to produce a slice that can be reproduced by another keypair.
```

Comment:
Maybe "the complementary keys"? It's a bit weird here because `self` is a KeyPair but only the PrivateKey part is used.